master/internal/api_user.go

```diff
@@ -683,3 +683,116 @@ func (a *apiServer) PostUserActivity(
 	}
 	return &apiv1.PostUserActivityResponse{}, err
 }
+
+func (a *apiServer) PostLongLivedToken(
+	ctx context.Context, req *apiv1.PostLongLivedTokenRequest,
+) (*apiv1.PostLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if err = user.AuthZProvider.Get().CanCreateUsersOwnToken(
+		ctx, *curUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	tokenExpiration := time.Now().Add(user.TokenExpirationDuration)
+	if req.Lifespan != "" {
+		d, err := time.ParseDuration(req.Lifespan)
+		if err != nil || d < 0 {
+			return nil, status.Error(codes.InvalidArgument,
+				"Lifespan must be a Go-formatted duration string with a positive value")
+		}
+		tokenExpiration = time.Now().Add(d)
+	}
+
+	token, err := user.DeleteAndCreateLongLivedToken(
```

Comment:
question (non-blocking): How do you feel about the use of `AccessToken` instead of LongLivedToken, it might be a more industry standard term that makes it clear what the purpose of these are.

Reply:
Lets wait for what product team thinks about it. I am open to be convinced either way.
Personally I like `LongLivedAccessToken` better.

Reply:
Will updated it based on the decisions made during the call with the product team.